[CI] XPackRestIT test {p0=license/20_put_license/Current license is trial means not eligle to start trial} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #9026 / ubuntu-2404-aarch64_checkrestcompat_platform-support-arm](https://gradle-enterprise.elastic.co/s/sr6o3wxogsioa)
- [elasticsearch-periodic-platform-support #8925 / ubuntu-2404-aarch64_checkrestcompat_platform-support-arm](https://gradle-enterprise.elastic.co/s/2b3d3h2wy4pca)

**Reproduction Line:**
```
./gradlew ":x-pack:qa:xpack-prefix-rest-compat:yamlRestTestV7CompatTest" --tests "org.elasticsearch.xpack.test.rest.XPackRestIT.test {p0=license/20_put_license/Current license is trial means not eligle to start trial}" -Dtests.seed=ADED095F269009D2 -Dtests.locale=bas-Latn-CM -Dtests.timezone=Asia/Taipei -Druntime.java=24
```

**Applicable branches:**
8.18

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.test.rest.XPackRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Bp0%3Dlicense%2F20_put_license%2FCurrent%20license%20is%20trial%20means%20not%20eligle%20to%20start%20trial%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [license/20_put_license:389]: expected [2xx] status code but api [xpack-license.get] returned [404 Not Found] [{ }
]
```

**Issue Reasons:**
- [8.18] 2 failures in test test {p0=license/20_put_license/Current license is trial means not eligle to start trial} (0.2% fail rate in 974 executions)
- [8.18] 2 failures in step ubuntu-2404-aarch64_checkrestcompat_platform-support-arm (12.5% fail rate in 16 executions)
- [8.18] 2 failures in pipeline elasticsearch-periodic-platform-support (12.5% fail rate in 16 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/qa/xpack-prefix-rest-compat/src/yamlRestTestV7Compat/java/org/elasticsearch/xpack/test/rest/XPackRestIT.java
+++ b/x-pack/qa/xpack-prefix-rest-compat/src/yamlRestTestV7Compat/java/org/elasticsearch/xpack/test/rest/XPackRestIT.java
@@ -10,6 +10,7 @@ package org.elasticsearch.xpack.test.rest;
 import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
 
 import org.elasticsearch.test.rest.yaml.ClientYamlTestCandidate;
+import org.junit.Before;
 
 public class XPackRestIT extends AbstractXPackRestTest {
 
@@ -20,5 +21,13 @@ public class XPackRestIT extends AbstractXPackRestTest {
     @ParametersFactory
     public static Iterable<Object[]> parameters() throws Exception {
         return createParameters();
+    }
+
+    /**
+     * Some rest tests depend on the trial license being generated before they run
+     */
+    @Before
+    public void setupLicense() {
+        super.waitForLicense();
     }
 }

